docs/RFCS/system_jobs.md

```diff
@@ -0,0 +1,421 @@
+```
+
+To update the status of a running job:
+
+```sql
+UPDATE system.jobs
+  SET percentCompleted = 24.42,
+      lastUpdated = now()
+```
+
+To mark a job as successful:
+
+```sql
+UPDATE system.jobs
+  SET percentCompleted = 100,
+      status = 'succeeded',
+      lastUpdated = now()
+```
+
+To mark a job as failed:
+
+```sql
+UPDATE system.jobs
+  SET status = 'failed',
+      lastUpdated = now(),
+      payload = {...payload, error = 's3.aws.amazon.com: host unreachable'}
+```
+
+To find queued or running jobs (e.g., for the default "System jobs" admin view):
+
+```sql
+SELECT * FROM system.jobs WHERE status IN ('pending', 'running');
+```
+
+In the above query, the admin interface will need to extract the `created` field
+from the `payload` protobuf in from each record in order to sort the records
+sensibly. It is currently impossible to do the sorting in the SQL CLI directly.
+(Similarly, it is not possible to limit the query to only jobs of a certain type
+without decoding the protobuf.)
+
+Under the proposed design, though, it's possible to retrieve the crucial
+progress information for a job with a known ID from the SQL CLI:
+
+```sql
+SELECT status, percentCompleted FROM system.jobs WHERE id = ?;
+```
+
+We currently have a customer requesting the ability to do at least the above
+from SQL. Additionally, support for reaching into protobuf columns from a SQL
+query, is planned.
+
+## Minor considerations
+
+- Cloud storage credentials can be passed in the query for `BACKUP` and
+  `RESTORE` commands. Care will be taken to strip those credentials from the
+  query text before the text is inserted into the `system.jobs` table.
+
+- Unlike past proposals, the schema intentionally lacks a `name` or
+  `description` column. As it stands, none of the job-creating queries allow the
+  user to specify such a name or description.
+
+# Drawbacks
+
+Requiring the job leader to periodically issue `UPDATE system.jobs SET
+percentCompleted = ...` queries to update the progress of running jobs is
+somewhat unsatisfying. One wishes to be able to conjure the `percentCompleted`
+column only when the record is read, but this design introduces significant
+implementation complexity.
+
+# Alternatives
+
+## Wider protobufs
+
+To further minimize the chances that we'll need to modify the `system.jobs`
+schema, we could instead stuff all the data into the `payload` protobuf:
+
+```sql
+CREATE TABLE system.jobs (
+    id      INT PRIMARY KEY,
+    jobType STRING NOT NULL,
+    payload BYTES,
+)
+```
+
+This allows for complete flexibility in adjusting the schema, but prevents
+essentially all useful SQL queries over the table until protobuf columns are
+natively supported.
+
+## Narrow protobufs
+
+We could also allow all data to be filtered by widening the `system.jobs` table
+to include all the fields proposed to be stored in the `payload` protobuf,
+potentially dropping `payload` entirely.
+
+```sql
+CREATE TABLE system.jobs (
```

Comment:
A more realistic option moves query + timestamps + error here, but leaves payload (with creator and the per-type payloads)